### PR TITLE
Add instructions for migrating away from log4j

### PR DESCRIPTION
**Pull request by jordansissel**
Target is for moving from SocketAppender to using filebeat for secure,
asynchronous, and reliable log transport.

### CODE REVIEW

File: lib/logstash/inputs/log4j.rb [hunk 1]
```diff
@@ -8,6 +8,78 @@
 require "timeout"
 require 'logstash-input-log4j_jars'
 
+# DEPRECATION NOTICE
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: `==== Deprecation Notice`

File: lib/logstash/inputs/log4j.rb [hunk 2]
```diff
@@ -8,6 +8,78 @@
...
+# ==================
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: remove this

File: lib/logstash/inputs/log4j.rb [hunk 3]
```diff
@@ -8,6 +8,78 @@
...
+# Configuring log4j for writing to local files
+# --------------------------------------------
+# 
+# In your log4j.properties file, remove SocketAppender and replace it with RollingFileAppender. 
+#
+# For example, you can use the following log4j.properties configuration to write daily log files.
+#
+#     # Your app's log4j.properties (log4j 1.2 only)
+#     log4j.rootLogger=daily
+#     log4j.appender.daily=org.apache.log4j.rolling.RollingFileAppender
```
Comment (jordansissel) [inline lib/logstash/inputs/log4j.rb]: RollingFileAppender in log4j v1 doesn't have any way to describe "Keep at most 7 old log files" so I don't know if this is the best recommendation. Users will need to configure log file cleanup externally.

File: lib/logstash/inputs/log4j.rb [hunk 4]
```diff
@@ -8,6 +8,78 @@
...
+#
+# The following section is a guide for how to migrate from SocketAppender to use filebeat.
+#
+# To migrate away from log4j SocketAppender to using filebeat, you will need to make 3 changes:
+#
+# 1) Configure your log4j.properties (in your app) to write to a local file.
+# 2) Install and configure filebeat to collect those logs and ship them to Logstash
+# 3) Configure Logstash to use the beats input.
+#
+# Configuring log4j for writing to local files
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: `.Paragraph title`

File: lib/logstash/inputs/log4j.rb [hunk 5]
```diff
@@ -8,6 +8,78 @@
...
+#     log4j.rootLogger=daily
+#     log4j.appender.daily=org.apache.log4j.rolling.RollingFileAppender
+#     log4j.appender.daily.RollingPolicy=org.apache.log4j.rolling.TimeBasedRollingPolicy
+#     log4j.appender.daily.RollingPolicy.FileNamePattern=/var/log/your-app/app.%d.log
+#     log4j.appender.daily.layout = org.apache.log4j.PatternLayout
+#     log4j.appender.daily.layout.ConversionPattern=%d{YYYY-MM-dd HH:mm:ss,SSSZ} %p %c{1}:%L - %m%n
+#
+# Configuring log4j.properties in more detail is outside the scope of this migration guide.
+#
+# Configuring filebeat
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: `.Paragraph title`

**Comment by jordansissel** [comment]
@suyograo updated based on your feedback. :)

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**